Packages/Tests/ApplicationFeatureTests/DataLoadingTests.swift

```diff
@@ -0,0 +1,111 @@
+@testable import AlchemyAPIClient
+@testable import ApplicationFeature
+import ComposableArchitecture
+import DataModel
+import XCTest
+
+@MainActor
```

Comment:
At one point I regretted adding this test, but forced myself through it 🙃. I was just thinking about how to cover the logic where app filtered list is not covering the screen and the app needs to keep pulling data